torch/_inductor/ir.py

```diff
@@ -748,32 +789,38 @@ def welford_combine_fn(
 
 @dataclasses.dataclass
 class Reduction(Loops):
-    reduction_ranges: List[Expr]
+    reduction_ranges: Sequence[_IntLike]
     reduction_type: str
     # self.dtype represents the dst dtype
     src_dtype: torch.dtype
     reduction_hint: ReductionHint
 
-    def __str__(self):
+    def __str__(self) -> str:  # type: ignore[override]
         return Loops.__str__(  # type: ignore[call-arg]
             self, names=("ranges", "reduction_ranges", "reduction_type")
         )
 
-    def __repr__(self):
+    def __repr__(self) -> str:  # type: ignore[override]
         return self.__str__()
 
-    def get_unbacked_symbol_uses(self) -> Set[sympy.Symbol]:
+    def get_unbacked_symbol_uses(self) -> Set[Symbol]:
         return super().get_unbacked_symbol_uses() | set().union(
             *(free_unbacked_symbols(e) for e in self.reduction_ranges)
         )
 
-    def get_reduction_size(self):
+    def get_reduction_size(self) -> Sequence[_IntLike]:
         return self.reduction_ranges
 
-    def get_reduction_type(self):
+    def get_reduction_type(self) -> Optional[str]:
         return self.reduction_type
 
-    def store_reduction(self, output_name, indexer, vars, reduction_vars):
+    def store_reduction(
+        self,
+        output_name: Optional[str],
+        indexer: Callable[[Sequence[Expr]], Never],
```

Comment:
What does the Never denote here ?

Reply:
I used `Never` to indicate un "unkown" type that I was hoping mypy would eventually resolve when we finally type the right set of functions (every test I tried set that to an empty array).

Reply:
It would actually mean the Callable doesn't return, like always raises an exception?